### PR TITLE
tgui sleepers; portable atmospherics and uplink search; POWER MONITOR; CARGO!

### LINKED ISSUE #15158: (usability issue) Ordering multiple cargo crates is horrible and inefficient.
We need a way to buy and authorize multiple crates at once, getting a bunch of crates one by one is incredibly slow and tedius when you're ordering a lot of them.

### PR DESCRIPTION
**Pull request by neersighted**
Also:
Fix some state logic
Fix SVGs
display: based filtering
input component
Move shuttle computers around: preparing to rewrite them
Generic "screen" capability: paving the way for paged UIs (air alarms use it now, cargo will one I convert it)

Closes #15158
:cl:
add: The Syndicate has stolen the latest tgui improvements from Nanotrasen! All uplinks now feature filtering/search.
add: Nanotrasen is proud to announce its next generation sleepers, featuring tgui!
add: Portable atmospheric components have been overhauled to use the latest interface technology.
del: The Area Atmosphere Computer has been removed.
add: Huge scrubbers now require power.
del: Borg jetpacks are now refilled from a recharger and not a canister.
add: Power monitors now have a state of the art graph. Hopefully they're actually useful to someone.
add: Cargo consoles have been redesigned and now feature search and a shopping cart!
/:cl:

Closes #15158
## Uplink

Colors
## Power Monitor

Heyo

Mouseover

### CODE REVIEW

**Comment by WJohn** [comment]
That search feature is AMAZING.

**Comment by ChangelingRain** [comment]
The syndicate UI appears to be using the wrong green for the eye and TC count; it looks like it's using `grass-green` and not `pale-green`

**Comment by xxalpha** [comment]
Nice.

**Comment by neersighted** [comment]
Here's what I'm working on now...

File: code/modules/mob/living/silicon/robot/robot_modules.dm [hunk 1]
```diff
@@ -99,11 +99,13 @@
 			var/obj/item/weapon/melee/baton/B = I
 			if(B.bcell)
 				B.bcell.charge = B.bcell.maxcharge
+		if(istype(I, /obj/item/weapon/tank/jetpack))
+			var/obj/item/weapon/tank/jetpack/J = I
+			J.air_contents.assert_gas("co2")
+			J.air_contents.gases["co2"][MOLES] = (6 * ONE_ATMOSPHERE) * J.volume / (R_IDEAL_GAS_EQUATION * T20C)
```
Comment (duncathan) [inline code/modules/mob/living/silicon/robot/robot_modules.dm]: Stealth

File: code/modules/atmospherics/machinery/components/unary_devices/thermomachine.dm [hunk 2]
```diff
@@ -6,6 +6,7 @@
 	var/icon_state_open = "cold_off"
 	density = TRUE
 	anchored = TRUE
+	idle_power_usage = 25
```
Comment (duncathan) [inline code/modules/atmospherics/machinery/components/unary_devices/thermomachine.dm]: Stealth

Comment (neersighted) [inline code/modules/atmospherics/machinery/components/unary_devices/thermomachine.dm]: Derp, was messing with power draw levels.

**Comment by neersighted** [comment]
Again

**Comment by GunHog** [comment]
GODLIKE.

**Comment by MrPerson** [comment]
The search is too jumpy. Ideally it would be a fixed width and not shrink slightly just because the longest name is no longer displayed.

**Comment by GunHog** [comment]
Is it possible to put the supply console in tabs? The list of crates can be very large, and might be helpful if they were tabbed or collapsible in some fasion.

File: code/controllers/subsystem/shuttles/supply.dm [hunk 3]
```diff
@@ -16,34 +16,53 @@
...
-		return forbidden_atoms_check(areaInstance)
+		return check_blacklist(areaInstance)
 	return ..()
 
-/obj/docking_port/mobile/supply/request(obj/docking_port/stationary/S)
+/obj/docking_port/mobile/supply/proc/check_blacklist(atom/A)
+	if(is_type_in_list(A, blacklist))
+		return 1
+	for(var/thing in A)
+		if(.(thing))
```
Comment (RemieRichards) [inline code/controllers/subsystem/shuttles/supply.dm]: Recursion isn't a good idea in byond due to proccall overhead.
I'd much prefer you make a list, containing all of A's contents (and their contents, and their contents, I think we have a proc for getting *ALL* contents) AND A itself, and then do the for loop on that, with a normal ```is_type_in_list()```

Comment (neersighted) [inline code/controllers/subsystem/shuttles/supply.dm]: This was existing code, I just moved it around. Agreed though, a better solution would be nice.

**Comment by neersighted** [comment]
@GunHog Not yet, though I'd like to add tabs down the line. Its just not quite doable at the moment. However, I think the search helps with this as you should really only see the list when browsing.

**Comment by Lumipharon** [comment]
these is some sexy graphs.
It would be awesome if the power monitor had a minimap like the crew monitor, only it showed the different areas of the ship, and you could colour them based off different settings like power consumption, power left in the apc, etc.

**Comment by Dorsisdwarf** [comment]
hngnngngnhnggh gnngnhg

you should do this for tablecraft sometime

File: code/datums/supplypacks.dm [hunk 4]
```diff
@@ -91,7 +91,6 @@ var/list/all_supply_groups = list(supply_emergency,supply_security,supply_engine
 	cost = 35
 	containertype = /obj/structure/closet/crate/internals
 	containername = "emergency crate"
-	group = supply_emergency
```
Comment (tkdrg) [inline code/datums/supplypacks.dm]: Why?

File: code/game/machinery/computer/buildandrepair.dm [hunk 5]
```diff
@@ -244,27 +241,10 @@
...
-			if("Switch to STANDARD","Switch to BROAD")
-				src.contraband_enabled = !src.contraband_enabled
-
-			if("Cancel")
-				return
-			else
-				user << "DERP! BUG! Report this (And what you were doing to cause it) to Agouri"
-	return
+		contraband = !contraband
+		user << "Receiver spectrum set to [contraband ? "Broad" : "Standard"]."
```
Comment (tkdrg) [inline code/game/machinery/computer/buildandrepair.dm]: `<span class='notice'>`

**Pull request merged**